### PR TITLE
PETSc fixes

### PR DESCRIPTION
**Pull request by connorjward**
# Description

Updates Firedrake to accept PETSc/SLEPc 3.22.2

Related PRs:

https://github.com/firedrakeproject/petsc/pull/24, https://github.com/firedrakeproject/slepc/pull/9

~Needs https://gitlab.com/petsc/petsc/-/issues/1678~

### CODE REVIEW

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:51]: Should this be removed now?

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:61]: Why are these being unset?

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:89]:
```suggestion
```
Once these have been merged.

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:64]: As above

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:93]: As above

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:101]: As above

File: .github/workflows/build.yml
Comment (JHopeCollins) [inline .github/workflows/build.yml:127]: As above

**Review by JHopeCollins — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (JHopeCollins) [reply .github/workflows/build.yml:61]: In which case, it all looks good to me. I can approve once the workflow is reverted and tests are passing.

Comment (connorjward) [reply .github/workflows/build.yml:61]: Now that I have some written approval I will probably just force things through (assuming things pass) so I don't need to bother you with extra work. Thanks!

**Comment by connorjward** [comment]
I want to run the tests one more time to make sure we haven't hit a performance regression. I have been keeping the runners busy today which might account for the slow CI time.

**Pull request merged**
